src/app/qgsvectorlayerproperties.cpp

```diff
@@ -531,6 +532,7 @@ void QgsVectorLayerProperties::apply()
     layer->setSubsetString( txtSubsetSQL->toPlainText() );
     mMetadataFilled = false;
   }
+  mOriginalSubsetSQL = txtSubsetSQL->toPlainText();
```

Comment:
I think it would be better to set it to layer->subsetString() instead of this, it's possible that the providers trim whitespaces or similar magic.